src/utils/hooks/contactsHooks.ts

```diff
@@ -1,21 +1,35 @@
-import { nameExistsInContacts } from "./contactsUtils";
+import { useGetOwnedAccountSafe } from "./getAccountDataHooks";
+import { isAddressValid } from "../../types/Address";
 import { useAppSelector } from "../redux/hooks";
+
 export const useContacts = () => useAppSelector(s => s.contacts);
 
 export const useAllSortedContacts = () => {
   const contacts = useContacts();
   return Object.values(contacts).sort((a, b) => a.name.localeCompare(b.name));
 };
 
-export const useContactExists = () => {
+export const useAddressExistsInContacts = () => {
   const contacts = useContacts();
-  return {
-    addressExistsInContacts: (pkh: string) => pkh in contacts,
-    nameExistsInContacts: (name: string) => nameExistsInContacts(contacts, name),
-  };
+  return (pkh: string) => pkh in contacts;
 };
 
 export const useGetContactName = () => {
   const contacts = useContacts();
   return (pkh: string) => (pkh in contacts ? contacts[pkh].name : undefined);
 };
+
+export const useValidatePkh = () => {
```

Comment:
it's better to be split into several validators. for example, if we use it in the address autocomplete as a destination for a tez transfer, why do we care if it exists in our accounts?
or, if it's intended to be used just for contacts, then let's call it `useValidateNewContactPkh` or something

Reply:
I think we only need a separate check for the contracts - renamed